modules/services/hypridle.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+
+  cfg = config.services.hypridle;
+in {
+  meta.maintainers = [ maintainers.khaneliman maintainers.fufexan ];
+
+  options.services.hypridle = {
+    enable = mkEnableOption "Hypridle, Hyprland's idle daemon";
+
+    package = mkPackageOption pkgs "hypridle" { };
+
+    settings = lib.mkOption {
+      type = with lib.types;
```

Comment:
Perhaps the type could be extracted into the hyprlang lib module and used from there throughout all hypr* modules.

Reply:
I do have a larger refactor of the hyprlang module in the hyprlock PR... https://github.com/nix-community/home-manager/pull/5325/files#diff-d125ebbed352e5f662545c5f6f0ef5d1f4b301290af30f43cc091069159b7f04 wondering if i should just consolidate all of these into a single PR

Reply:
Implemented this in all 3 prs